File: src/python/2401_2700/2643_row_with_maximum_ones.py
from typing import List


def rowAndMaximumOnes(mat: List[List[int]]) -> List[int]:
    """Return the index of the row with the maximum number of ones."""
    res = [0, 0]
    for i, row in enumerate(mat):
        cnt = sum(row)
        if cnt > res[1]:
            res[0], res[1] = i, cnt

    return res


if __name__ == "__main__":
    mat = [[0, 0, 0, 1], [0, 1, 1, 1], [1, 1, 1, 1], [0, 0, 0, 0]]
    print(rowAndMaximumOnes(mat))  # [2, 4]

File: leetpattern/cpp/2401_2700/2643_row_with_maximum_ones.cc
#include <iostream>
#include <numeric>
#include <vector>
using namespace std;

vector<int> rowAndMaximumOnes(vector<vector<int>>& mat) {
    vector<int> res = {0, 0};

    for (size_t i = 0; i < mat.size(); i++) {
        int cnt = accumulate(mat[i].begin(), mat[i].end(), 0);
        if (cnt > res[1]) {
            res[0] = (int)i;
            res[1] = cnt;
        }
    }
    return res;
}

int main() {
    vector<vector<int>> mat = {
        {0, 0, 0, 1}, {0, 1, 1, 1}, {1, 1, 1, 1}, {0, 0, 0, 0}};
    vector<int> res = rowAndMaximumOnes(mat);
    cout << res[0] << ", " << res[1] << endl;  // 2, 4
    return 0;
}
